### ISSUE TITLE
`prefer-t-regex` crashes ESLint

### ISSUE THREAD

**Issue by ehmicky** [avajs/eslint-plugin-ava#251]
`example.js`:

```js
import test from 'ava'

const func = () => {}

test('example', t => {
  t.true(func())
})
```

`.eslintrc.yml`:

```yml
parserOptions:
  ecmaVersion: 2019
  sourceType: module
plugins: [eslint-plugin-ava]
rules:
  ava/prefer-t-regex: 2
```

`package.json`:

```json
{
  "name": "eslint-plugin-ava-bug",
  "version": "0.0.1",
  "dependencies": {
    "eslint": "^5.16.0",
    "eslint-plugin-ava": "^7.0.0"
  }
}
```

Then:

```
$ eslint example.js 
TypeError: Cannot destructure property `name` of 'undefined' or 'null'.
Occurred while linting /home/ether/Desktop/eslint-plugin-ava-bug/example.js:6
    at /home/ether/Desktop/eslint-plugin-ava-bug/node_modules/eslint-plugin-ava/rules/prefer-t-regex.js:44:10
    at /home/ether/Desktop/eslint-plugin-ava-bug/node_modules/enhance-visitors/index.js:25:12
    at /home/ether/Desktop/eslint-plugin-ava-bug/node_modules/enhance-visitors/index.js:15:7
    at /home/ether/Desktop/eslint-plugin-ava-bug/node_modules/eslint/lib/util/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/ether/Desktop/eslint-plugin-ava-bug/node_modules/eslint/lib/util/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/ether/Desktop/eslint-plugin-ava-bug/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/home/ether/Desktop/eslint-plugin-ava-bug/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (/home/ether/Desktop/eslint-plugin-ava-bug/node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (/home/ether/Desktop/eslint-plugin-ava-bug/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:632:23)
```

Node `12.3.1`
Ubuntu `19.04`

**Comment by novemberborn** [comment]
That looks like this line:

https://github.com/avajs/eslint-plugin-ava/blob/32b45c05ab5b505dc3ea84a52949042446081fe3/rules/prefer-t-regex.js#L44

I guess the assumption there is wrong. @GMartigny what do you reckon?

**Comment by GMartigny** [comment]
You're absolutely right @novemberborn.
I did not consider this case. I pushed a fix with regression test.

**Comment by ehmicky** [comment]
I just checked it and it works, thanks @GMartigny and @novemberborn!

**Comment by novemberborn** [comment]
Thanks @ehmicky! I’m working on a fix for #252 as well. Will ping you once that’s ready.
